raiden/network/pathfinding.py

```diff
@@ -42,6 +45,66 @@
 log = structlog.get_logger(__name__)
 
 
+@dataclass(frozen=True)
+class PFSInfo:
+    url: str
+    price: TokenAmount
+    chain_id: ChainID
+    token_network_registry_address: TokenNetworkAddress
+    payment_address: Address
+    message: str
+    operator: str
+    version: str
+    settings: str
+
+
+@dataclass
+class PFSConfig:
+    info: PFSInfo
+    maximum_fee: TokenAmount
+    iou_timeout: BlockNumber
+    max_paths: int
+
+
+@dataclass
+class IOU:
+    sender: Address
+    receiver: Address
+    one_to_n_address: Address
+    amount: TokenAmount
+    expiration_block: BlockNumber
+    chain_id: ChainID
+    signature: Optional[Signature] = None
+
+    def sign(self, privkey: bytes) -> None:
+        self.signature = Signature(
+            sign_one_to_n_iou(
+                privatekey=encode_hex(privkey),
+                sender=to_checksum_address(self.sender),
+                receiver=to_checksum_address(self.receiver),
+                amount=self.amount,
+                expiration_block=self.expiration_block,
+                one_to_n_address=to_checksum_address(self.one_to_n_address),
+                chain_id=self.chain_id,
+            )
+        )
+
+    def as_json(self) -> Dict[str, Any]:
```

Comment:
Maybe this could be on module level instead, since we're generally avoiding adding conversion methods to dataclasses. Although this one does not belong to the state machine dataclasses, so it's not that important here.

Reply:
If it's OK for you I'd leave it as is right now. Ideally this should probably use the serialization infrastructure as well.

Reply:
Ok, let's leave it like that for now.